torch/distributed/optim/named_optimizer.py

```diff
@@ -170,10 +189,9 @@ def load_state_dict(self, state_dict: Mapping[str, Any]) -> None:
                 f"Expects equal length as {len(new_state)} in `state_dict` state length but found {len(state)}."
             )
         for idx, param_key in enumerate(self.ordered_param_keys):
+            # When the conditional training is performed, not all parameters are updated in the optim.
```

Comment:
Can we add a parity test with regular optimizer under conditional training use case?

Reply:
Done, added in https://github.com/pytorch/pytorch/pull/91344